### ISSUE TITLE
What event struct should I be using for CloudTrail?

### ISSUE THREAD

**Issue by kaihendry** [aws/aws-lambda-go#80]
My use case is that I want to post CloudTrail events to my company's Slack so I can see who is logging in etc. I think this is possible from Creating a Trail and assigning to my Golang function, e.g.

<img src="https://s.natalian.org/2018-05-18/1526612801_2558x1406.png">

I can't seem to see a corresponding struct to map this event onto:
https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-event-reference-aws-console-sign-in-events.html

**Comment by mweagle** [comment]
Hi 👋. Following up on this to see if there are plans to represent CloudTrail events with a canonical type. Thanks!

**Comment by Tarasovych** [comment]
I'd like to see event struct for IAM events as well

**Comment by pescetto** [comment]
Yeah, I'm trying to implement this eventbridge for [EKS Node Group creation](https://docs.aws.amazon.com/eks/latest/userguide/enable-asg-metrics.html) to enable metrics, but there's no CloudTrail event type. Not sure if I need to marshal it myself. I don't see any PR's for this. Would be nice to have it supported.
